### PR TITLE
change apps/v1beta2 StatefulSet observedGeneration from a pointer to an int for consistency

### LINKED ISSUE #49623: change apps/v1beta2 StatefulSet observedGeneration from a pointer to an int for consistency
/kind cleanup
/sig apps
/area workload-api

In apps/v1beta2, we need to change the StatefulSet observedGeneration from a pointer to an int for consistency as listed in #49135.

/cc @janetkuo @kow3ns

### PR DESCRIPTION
**Pull request by dixudx**
**What this PR does / why we need it**:
change the StatefulSet observedGeneration from a pointer to an int for consistency

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #49623
xref #49135

**Special notes for your reviewer**:
/cc @janetkuo @foxish @kow3ns 

**Release note**:

```release-note
change apps/v1beta2 StatefulSet observedGeneration (optional field) from a pointer to an int for consistency
```

### CODE REVIEW

File: staging/src/k8s.io/api/apps/v1beta2/types.go [hunk 1]
```diff
@@ -225,7 +225,7 @@ type StatefulSetStatus struct {
 	// observedGeneration is the most recent generation observed for this StatefulSet. It corresponds to the
 	// StatefulSet's generation, which is updated on mutation by the API Server.
 	// +optional
-	ObservedGeneration *int64 `json:"observedGeneration,omitempty" protobuf:"varint,1,opt,name=observedGeneration"`
+	ObservedGeneration int64 `json:"observedGeneration,omitempty" protobuf:"varint,1,opt,name=observedGeneration"`
```
Comment (wojtek-t) [inline staging/src/k8s.io/api/apps/v1beta2/types.go]: Can we do this? Isn't it a backward incompatible change?
@deads2k @smarterclayton

Comment (deads2k) [inline staging/src/k8s.io/api/apps/v1beta2/types.go]: > Can we do this? Isn't it a backward incompatible change?
> @deads2k @smarterclayton

Isn't the `v1beta2` serialization net new in 1.8?  If it is, then the proto serialization won't be a problem.  Not sure how you'd roundtrip through it though.  The compatibility of that bit depends on the meaning of nil versus 0 in the API itself.

Comment (janetkuo) [inline staging/src/k8s.io/api/apps/v1beta2/types.go]: Yes, apps/v1beta2 is new and WIP so it's safe to break it.

**Comment by mbohlool** [comment]
/unassign mbohlool

Looks good from API point of view (backward compatibility is not an issue here).

**Review by kow3ns — APPROVED** [review]

**Comment by kow3ns** [comment]
@dixudx You will probably need to update the fuzzer to support round trip conversions from a pointer to a integer value and vice versa.

**Comment by dixudx** [comment]
@janetkuo @foxish @wojtek-t Need your `approval`. Thanks.

**Comment by kow3ns** [comment]
@dixudx One thing, can you update the release note to indicate that the observedGeneration field of the StatefulSetStatus Kind in the apps/v1beta2 group is an optional integer and not a pointer.

**Review by kow3ns — APPROVED** [review]

**Comment by dixudx** [comment]
@wojtek-t Need your `lgtm` also. Thanks.

**Comment by dixudx** [comment]
@kow3ns Done.

**Pull request merged**